extensions/skyportal/skyportal/handlers/api/alert.py

```diff
@@ -323,6 +377,30 @@ def get(self, objectId: str = None):
               application/json:
                 schema: Error
         """
+        single_user_group = (
+            DBSession().query(Group).filter(Group.name == self.current_user.username).first()
+        )
+        # grab streams accessible to user:
+        if single_user_group is not None:
+            streams = (
+                DBSession()
+                .query(Stream)
+                .join(GroupStream)
+                .filter(GroupStream.group_id == single_user_group.id)
+                .all()
+            )
+        else:
+            streams = []
+
+        # allow access to public data only by default
+        selector = {1}
+
+        for stream in streams:
+            if "ztf" in stream.name.lower():
+                selector.update(set(stream.altdata.get("selector", [])))
+
+        selector = list(selector)
```

Comment:
This pattern is repeated a lot, can it be refactored into a function ?

Reply:
Refactrored